src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_validators.py

```diff
@@ -64,8 +64,8 @@ def internal_validate_lock_parameters(resource_group_name, resource_provider_nam
 
 
 def validate_lock_parameters(namespace):
-    internal_validate_lock_parameters(namespace.get('resource_group_name', None),
-                                      namespace.get('resource_provider_namespace', None),
-                                      namespace.get('parent_resource_path', None),
-                                      namespace.get('resource_type', None),
-                                      namespace.get('resource_name', None))
+    internal_validate_lock_parameters(namespace.resource_group_name,
```

Comment:
This will result in another failure if the namespace doesn't have those parameters.  Instead try:

getattr(namespace, 'resource_group_name', None)

This will work and will do what you were trying to accomplish with the namespace.get